hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/DirectWriteMarkers.java

```diff
@@ -91,22 +98,10 @@ public boolean doesMarkerDirExist() throws IOException {
 
   @Override
   public Set<String> createdAndMergedDataPaths(HoodieEngineContext context, int parallelism) throws IOException {
-    Set<String> dataFiles = new HashSet<>();
-
-    List<StoragePathInfo> topLevelInfoList = storage.listDirectEntries(markerDirPath);
-    List<String> subDirectories = new ArrayList<>();
-    for (StoragePathInfo topLevelInfo: topLevelInfoList) {
-      if (topLevelInfo.isFile()) {
-        String pathStr = topLevelInfo.getPath().toString();
-        if (pathStr.contains(HoodieTableMetaClient.MARKER_EXTN) && !pathStr.endsWith(IOType.APPEND.name())) {
-          dataFiles.add(translateMarkerToDataPath(pathStr));
-        }
-      } else {
-        subDirectories.add(topLevelInfo.getPath().toString());
-      }
-    }
-
-    if (!subDirectories.isEmpty()) {
+    Pair<List<String>, Set<String>> subDirectoriesAndDataFiles = getSubDirectoriesByMarkerCondition(storage.listDirectEntries(markerDirPath), NOT_APPEND_MARKER_PREDICATE);
```

Comment:
If the `APPEND` marker is only used for table version 6, create `DirectWriteMarkersV1` with `APPEND` and `DirectWriteMarkersV2` without `APPEND`.

Reply:
Fixed by introducing a new `DirectWriteMarkersV1 ` implementation class that implements `AppendMarkerHandler`.